winter_messaging_outbox_transacton/timeout_handler.py

```diff
@@ -0,0 +1,39 @@
+import signal
+from functools import wraps
+
+
+class TimeoutException(Exception):
+    pass
+
+
+def raise_timeout_exception(signum, frame):
+    raise TimeoutException()
+
+
+class TimeoutHandler:
+    can_retry = True
+
+    @staticmethod
+    def timeout(seconds: int, retries: int = 0):
+        def _decorator(func):
+            @wraps(func)
+            def decorated_func(*args, **kwargs):
+                attempts = 0
+                while True:
+                    signal.signal(signal.SIGALRM, raise_timeout_exception)
+                    signal.setitimer(signal.ITIMER_REAL, seconds)
+                    try:
+                        func(*args, **kwargs)
```

Comment:
I think it is better to use

```
return func(*args, **kwargs)
```
and delete else block (I think we do not usually use else in try/except, because sometimes it is hard to understand it for some people, and it is better to delete it if it is possible)

or 

```
func(*args, **kwargs)
return 
```
if we really want to return None always

Reply:
Also it is better to use for instead of while

```
for attempts in range(retries):
     ...
     return 
```